javascriptv3/example_code/iam/scenarios/src/iam_basics.js

```diff
@@ -0,0 +1,410 @@
+              );
+
+              const assume_policy_arn = data.Policy.Arn;
+              try {
+                console.log("\nAttaching the policy to the user.....\n");
+                const attach_policy_to_user_params = {
+                  PolicyArn: assume_policy_arn,
+                  UserName: user_name,
+                };
+                const data = await iamClient.send(
+                    new AttachUserPolicyCommand(attach_policy_to_user_params)
+                );
+                console.log(
+                    "\nWaiting 10 seconds for policy to be attached...\n"
+                );
+                wait(10000);
+                console.log(
+                    "Success. Policy attached to user " + user_name + "."
+                );
+                try {
+                  console.log(
+                      "\nAssume for the user the role with permission to list all buckets....\n"
+                  );
+                  const assume_role_params = {
+                    RoleArn: role_arn, //ARN_OF_ROLE_TO_ASSUME
+                    RoleSessionName: "session1",
+                    DurationSeconds: 900,
+                  };
+                  // Create Amazon STS client with the credentials for the user. Remember, the user has permissions to assume roles using Amazon STS.
+                  const stsClientWithUsersCreds = new STSClient({
+                    credentials: user_creds,
+                    region: REGION,
+                  });
+
+                  const data = await stsClientWithUsersCreds.send(
+                      new AssumeRoleCommand(assume_role_params)
+                  );
+                  console.log(
+                      "Success assuming role. Access key id is " +
+                      data.Credentials.AccessKeyId +
+                      "\n" +
+                      "Secret access key is " +
+                      data.Credentials.SecretAccessKey
+                  );
+
+                  const newAccessKey = data.Credentials.AccessKeyId;
+                  const newSecretAccessKey = data.Credentials.SecretAccessKey;
+
+                  console.log(
+                      "\nWaiting 10 seconds for the user to assume the role with permission to list all buckets...\n"
+                  );
+                  wait(10000);
+                  // Set the parameters for the temporary credentials, which gives permission to list S3 buckets.
+                  var new_role_creds = {
+                    accessKeyId: newAccessKey,
+                    secretAccessKey: newSecretAccessKey,
+                    sessionToken: data.Credentials.SessionToken,
+                  };
+                  try {
+                    console.log(
+                        "Listing the S3 buckets using the credentials of the assumed role... \n"
+                    );
+                    // Create and S3 client with the temporary credentials.
```

Comment:
Replace "and" with "an": Create an S3 client...

Reply:
done